src/ChannelServer/World/Character.cs

```diff
@@ -306,251 +306,1830 @@ public int ClassLevel
 		public int TotalExp { get; set; }
 
 		/// <summary>
+		/// Gets or set Job HpRate
+		/// </summary>
+		public float HpRateByJob { get; set; }
+
+		/// <summary>
 		/// Gets or set HP, clamped between 0 and MaxHp.
 		/// </summary>
 		public int Hp
 		{
 			get { return _hp; }
 			set { _hp = Math2.Clamp(0, this.MaxHp, value); }
 		}
-		private int _hp;
+		private int _hp;
+
+		/// <summary>
+		/// Maximum HP.
+		/// </summary>
+		public int MaxHp
+		{
+			get
+			{
+				var level = this.Level;
+				var rateByJob = this.HpRateByJob;
+				var stat = this.Con;
+
+				var byJob = Math.Floor(400 * rateByJob);
+				var byLevel = Math.Floor(byJob + ((level - 1) * 80 * rateByJob));
+				var byStat = Math.Floor(((stat * 0.005f) + (Math.Floor(stat / 10.0f) * 0.015f)) * byLevel);
+				var byItem = 0; // TODO: "MHP"
+
+				// Buffs: "MHP_BM"
+				var byBuffs = 0;
+
+				// "MHP_Bonus"
+				var byBonus = 0;
+
+				var value = byLevel + byStat + byItem + byBuffs + byBonus;
+
+				return (int)value;
+			}
+		}
+
+		/// <summary>
+		/// Gets or set Job SpRate
```

Comment:
This comment could be better, what exactly is the "SP Rate"? Think about what you would want to know about that property when writing the comment.